src/main/java/sirius/kernel/commons/Strings.java

```diff
@@ -381,15 +381,61 @@ public static String limit(@Nullable Object input, int length) {
      * than <tt>length</tt>, the full value is returned. If input is <tt>null</tt>, "" is returned.
      */
     public static String limit(@Nullable Object input, int length, boolean showEllipsis) {
+        return limit(input, length, showEllipsis, false, 0);
+    }
+
+    /**
+     * Limits the length of the given string to the given length.
+     *
+     * @param input the object which string representation should be limited to the given length
+     * @param length the max. number of characters to return
+     * @param showEllipsis whether to append three dots if <tt>input</tt> is longer than <tt>length</tt>
+     * @param showTruncated whether to append a marker/signal if <tt>input</tt> is longer than <tt>length</tt>
+     * @param charactersToPreserveAtTheEnd the number of characters to preserve at the end of the string
+     * @return a part of the string representation of the given <tt>input</tt>. If input is shorter
+     * than <tt>length</tt>, the full value is returned. If input is <tt>null</tt>, "" is returned.
+     * If <tt>charactersToPreserveAtTheEnd</tt> is negativ it is omitted.
+     */
+    public static String limit(@Nullable Object input,
+                               int length,
+                               boolean showEllipsis,
+                               boolean showTruncated,
+                               int charactersToPreserveAtTheEnd) {
```

Comment:
Do all combinations of these parameters actually make sense? Wouldn't it be clearer if the "truncate in the middle" function would be separate instead of cramming it into the limit function?

Can be quite confusion to set them up to make exactly what you want.
If we split them we could also provide some small examples in the JavaDoc :)